### ISSUE TITLE
Support custom 'CommentChar'

### ISSUE THREAD

**Issue by gustavokrieger** [fsmiamoto/git-todo-parser#5]
The todo parser fails when the 'CommentChar' has been set to something other than '#' (by running `git config --global core.commentChar ";"`, for example). It seems that there is no way to pass a custom one to the parser right now, since it is [hard-coded](https://github.com/fsmiamoto/git-todo-parser/blob/main/todo/todo.go#L26).

The necessity for this came from the Lazygit repo. It started at https://github.com/jesseduffield/lazygit/issues/2601 and was further discussed at https://github.com/jesseduffield/lazygit/pull/2639.

**Comment by gustavokrieger** [comment]
@fsmiamoto, what do you think?

**Comment by stefanhaller** [comment]
We have two options to solve this:
1. add a `commentChar` parameter to `Parse()` and `Write()`. That's a breaking change, but a trivial one, and since we are at a `0.x` major version, I'd say we don't have to worry about API stability.
2. keep the existing `Parse()` and `Write()` functions the same, but add new ones that take the comment char. (Implement the existing ones in terms of the new ones, passing `"#"`.)

I'm not sure how many down-stream clients we have, or how to find out. I'd be surprised if there are many. Personally I'd vote for 1., since it's simpler.

**Comment by gustavokrieger** [comment]
Since there may be other existing/future config options that will also need to be passed in, how about having a config struct similar to this:

```go
type Config struct {
	Core core
}

type core struct {
	CommentChar string
}

func NewConfig() Config {
	return Config{
		Core: core{
			CommentChar: "#",
		},
	}
}

func Parse(f io.Reader, conf Config) ([]Todo, error) {
```

As an interesting non-breaking alternative, there is this ["Variadic options"](https://google.github.io/styleguide/go/best-practices#variadic-options) from the google styleguide.

**Comment by stefanhaller** [comment]
I don't see what other options we could need to pass in, so my preference would be to keep it simple and only pass in the commentChar. We could still refactor this into a struct later when it turns out that we need it. Yes, that would be another breaking change then, but again a simple one for clients to adapt to. I'm always a fan of introducing complexity only when you need it.

The variadic options pattern is a smart idea, but definitely overkill for a simple case like this.

Since we don't seem to be getting input from @fsmiamoto, I'd encourage you to go ahead with this and make a PR. I have merge rights here, so I can see it through. @fsmiamoto, please do let me know if you have objections.

**Comment by stefanhaller** [comment]
Addressed by #6.

**Comment by fsmiamoto** [comment]
Sorry for the delay guys, I was traveling in the last few weeks so couldn't keep up.
In anyway, I'm onboard with the changes made so great job!
